### PR TITLE
Update haiku_generator to use generative AI API to write haikus

### PR DESCRIPTION
**Pull request by windmaple**
*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. For larger changes, raising an issue first helps
reduce redundant work.*

## Pre-launch Checklist

- [ ] I read the [Effective Dart: Style] _recently_, and have followed its advice.
- [ ] I signed the [CLA].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Effective Dart: Style]: https://dart.dev/guides/language/effective-dart/style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: haiku_generator/finished/lib/data/repositories/poem_repository_impl.dart [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import '../../domain/repositories/abstract/poem_repository.dart';
+import 'dart:convert';
+import 'package:http/http.dart' as http;
+
+class PoemRepositoryImpl implements PoemRepository {
+  PoemRepositoryImpl();
+
+  @override
+  Future<String> getPoems(String productName) async {
+    // Replace YOUR_API_KEY with your API key.
```
Comment (domesticmouse) [inline haiku_generator/finished/lib/data/repositories/poem_repository_impl.dart:10]: Nit: mark this as a `// TODO`

**Review by domesticmouse — APPROVED** [review]
Please consider rebuilding the running with Flutter 3.10 beta, and changing the Dart minimum version in `pubspec.yaml` as `sdk: ^3.0.0-0`

File: haiku_generator/finished/lib/data/repositories/poem_repository_impl.dart (hunk 1 above)
Comment (windmaple) [reply haiku_generator/finished/lib/data/repositories/poem_repository_impl.dart]: done

**Review by domesticmouse — APPROVED** [review]

**Comment by domesticmouse** [comment]
You'll need to comment `haiku_generator` out of the stable CI in https://github.com/windmaple/flutter-codelabs/blob/main/flutter_ci_script_stable.sh

Add a TODO for me to add it back in on stable version increment and I'll make sure it happens in https://github.com/flutter/codelabs/pull/1577

**Pull request merged**